### DISCUSSION TITLE
Deploying the databasegpt.web on localhost for testing

### DISCUSSION THREAD

**Discussion by suntsu1970JP** [marcominerva/DatabaseGPT#76] [Q&A] [answered]
Hello Marco,
Before anything else, the application is excellent, and you did a great job. Anyway, I could run the web version of the program using Visual Studio 2022. I installed the .NET 8.0 SDK bundle (including hosting and ASP.NET Core SDK) and generated the OpenAI key, following all the instructions on the website. However, when I tried to deploy the application—either by publishing it to a file or using the web server deploy feature on my IIS localhost (Windows 11)—I received an error stating that it could not reach the page. Is there anything I might have missed?  Your help is highly appreciated.

Regards,
Sunz JP

**Comment by marcominerva** [comment]
Have you done all the required configuration? You could check for example this video to be sure you didn't forget anything: https://www.youtube.com/watch?v=OlkpRoE2mYQ.

**Reply by suntsu1970JP** [reply]
Yes, I watched that video. I have already installed the necessary SDK and hosting bundles. The only difference from the video is that the person deployed it on a server, while I deployed it on my localhost, which runs Windows 11. Is there any difference between the two setups? Have you attempted to deploy this on any platform other than the Azure Server? I also noticed an error when I was creating the published settings. Do you have any recommendations?
Here's the screenshot,
![ErrorMigration](https://github.com/user-attachments/assets/e26966c7-6bf7-4754-a603-a0e93c546c0b)

![Error](https://github.com/user-attachments/assets/403ea78a-7e1f-455a-9a8a-be56ca7f3b5c)

**Reply by marcominerva** [reply, marked as answer]
Do Other deployed sites works correctly? Do you have any erorr in the Windows Event Log?

**Reply by suntsu1970JP** [reply]
Thanks I solved the issue. It is the folder restriction.

Here's the error
**_Category: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware
EventId: 1
SpanId: fb230cb5e3e93b98
TraceId: 79a7f0e5ef818ee8cd0a7eb801a8e749
ParentId: 0000000000000000
RequestId: 4000000c-0000-fd00-b63f-84710c7967bb
RequestPath: /ChatDB/lib/bootstrap/css/bootstrap.min.css

An unhandled exception has occurred while executing the request.

Exception: 
System.UnauthorizedAccessException: Access to the path 'C:\inetpub\wwwroot\ChatDB\obj\WebOptimizerCache' is denied.
   at System.IO.FileSystem.CreateDirectory(String fullPath, Byte[] securityDescriptor)
   at System.IO.Directory.CreateDirectory(String path)
   at WebOptimizer.AssetResponseStore.AddAsync(String bucket, String cachekey, AssetResponse assetResponse)
   at WebOptimizer.AssetBuilder.BuildAsync(IAsset asset, HttpContext context, IWebOptimizerOptions options)
   at WebOptimizer.AssetMiddleware.HandleAssetAsync(HttpContext context, IAsset asset, WebOptimizerOptions options)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.<Invoke>g__Awaited|10_0(ExceptionHandlerMiddlewareImpl middleware, HttpContext context, Task task)_**
